torch/nn/parallel/distributed.py

```diff
@@ -541,6 +542,12 @@ def train(self, mode=True):
         for module in self._module_copies[1:]:
             module.train(mode)
 
+    # Communication hook is an enhancement that provides a hook which can be
+    # used to implement various Gradient Compression algorithms.
+    def register_comm_hook(self, state: object, hook: callable):
```

Comment:
Since this is a public-facing API, we should have documentation and a simple example of how to use it in the docstrings for this function (you can see some examples in the RPC codebase such as https://github.com/pytorch/pytorch/blob/master/torch/distributed/rpc/api.py#L756). (If the hook isn't fully functional from a user perspective just yet, then we can mark the API as private (leading underscore) and then mark it as public when the API is complete.)

Reply:
Thanks for pointing out @rohan-varma , I included a documentation. It is ready to review. One todo item: we may need an additional example for state object use case.